src/express-lane-auction/ExpressLaneAuction.sol

```diff
@@ -0,0 +1,538 @@
+        // additional custom admin role for each of them to allow for key rotation management
+        setRoleAndAdmin(
+            AUCTIONEER_ROLE,
+            args._auctioneer,
+            AUCTIONEER_ADMIN_ROLE,
+            args._auctioneerAdmin
+        );
+        setRoleAndAdmin(
+            RESERVE_SETTER_ROLE,
+            args._reservePriceSetter,
+            RESERVE_SETTER_ADMIN_ROLE,
+            args._reservePriceSetterAdmin
+        );
+        _grantRole(ROUND_TIMING_SETTER_ROLE, args._roundTimingSetter);
+    }
+
+    /// @notice Set an address for a role, an admin role for the role, and an address for the admin role
+    function setRoleAndAdmin(
+        bytes32 role,
+        address roleAddr,
+        bytes32 adminRole,
+        address adminRoleAddr
+    ) internal {
+        _grantRole(role, roleAddr);
+        _grantRole(adminRole, adminRoleAddr);
+        _setRoleAdmin(role, adminRole);
+    }
+
+    /// @inheritdoc IExpressLaneAuction
+    function currentRound() external view returns (uint64) {
+        return roundTimingInfo.currentRound();
+    }
+
+    /// @inheritdoc IExpressLaneAuction
+    function isAuctionRoundClosed() external view returns (bool) {
+        return roundTimingInfo.isAuctionRoundClosed();
+    }
+
+    /// @inheritdoc IExpressLaneAuction
+    function isReserveBlackout() external view returns (bool) {
+        (ELCRound storage lastRoundResolved, ) = latestResolvedRounds.latestELCRound();
+        return roundTimingInfo.isReserveBlackout(lastRoundResolved.round);
+    }
+
+    /// @inheritdoc IExpressLaneAuction
+    function roundTimestamps(uint64 round) external view returns (uint64, uint64) {
+        return roundTimingInfo.roundTimestamps(round);
+    }
+
+    /// @inheritdoc IExpressLaneAuction
+    function setBeneficiary(address newBeneficiary) external onlyRole(BENEFICIARY_SETTER_ROLE) {
+        emit SetBeneficiary(beneficiary, newBeneficiary);
+        beneficiary = newBeneficiary;
+    }
+
+    function _setReservePrice(uint256 newReservePrice) private {
+        if (newReservePrice < minReservePrice) {
+            revert ReservePriceTooLow(newReservePrice, minReservePrice);
+        }
+
+        emit SetReservePrice(reservePrice, newReservePrice);
+        reservePrice = newReservePrice;
+    }
+
+    // CHRIS: TODO: tests
+    /// @inheritdoc IExpressLaneAuction
+    function setRoundTimingInfo(RoundTimingInfo calldata newRoundTimingInfo)
+        external
+        onlyRole(ROUND_TIMING_SETTER_ROLE)
+    {
+        RoundTimingInfo memory currentRoundTimingInfo = roundTimingInfo;
+
+        uint64 currentCurrentRound = currentRoundTimingInfo.currentRound();
+        uint64 newCurrentRound = newRoundTimingInfo.currentRound();
+        // updating round timing info needs to be synchronised
+        // so we ensure that the current round won't change
+        if (currentCurrentRound != newCurrentRound) {
```

Comment:
In some scenerio it would not be possible to configure a offset such that `currentCurrentRound == newCurrentRound`, e.g. if round duration is 1 second and 20000 round has passed then it is not possible to set the round duration to 86400 because offset would have to be negative. I think it's ok to be a won't fix or we can also hardcode a minimum round duration